cloner/cache_cloner.go

```diff
@@ -0,0 +1,43 @@
+package cloner
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/drone/plugin/cache"
+	"golang.org/x/exp/slog"
+)
+
+func NewCache(cloner Cloner) *cacheCloner {
+	return &cacheCloner{cloner: cloner}
+}
+
+type cacheCloner struct {
+	cloner Cloner
+}
+
+func (c *cacheCloner) Clone(ctx context.Context, repo, ref, sha string) (string, error) {
+	key := cache.GetKeyName(fmt.Sprintf("%s%s%s", repo, ref, sha))
```

Comment:
Please add comments for what the functions are doing giving an example repo clone.

Reply:
Added comments